### PR TITLE
Handle application/octet-stream in verification-client

### PR DESCRIPTION
**Pull request by robert3005**
## Before this PR
We would try to deserialize application/octet-stream as json

## After this PR
==COMMIT_MSG==
Don't try to deserialize applicaiton/octet-stream and handle it as a binary stream
==COMMIT_MSG==

## Possible downsides?
N/A - this was missing functionality

### CODE REVIEW

**Comment by robert3005** [comment]
Tested against palantir/conjure-java-runtime and the binary client tests pass with this change

File: verification-client/src/resource.rs [hunk 1]
```diff
@@ -158,7 +161,7 @@ impl VerificationClientResource {
         let content_type = response
```
Comment (ferozco) [inline verification-client/src/resource.rs:161]: a content type should always be set, we should fail here if its not. It should also make the rest of the code cleaner

Comment (robert3005) [reply verification-client/src/resource.rs:161]: not that much but can unwrap the optional as well

Comment (robert3005) [reply verification-client/src/resource.rs:161]: we do not send content-type in 204 responses

**Comment by ferozco** [comment]
:+1:

**Pull request merged**